x-pack/plugins/ingest_manager/server/services/epm/packages/install.ts

```diff
@@ -326,3 +341,32 @@ export async function ensurePackagesCompletedInstall(
   await Promise.all(installingPromises);
   return installingPackages;
 }
+
+export function getInstallType({
+  pkgVersion,
+  installedPkg,
+}: {
+  pkgVersion: string;
+  installedPkg: SavedObject<Installation> | undefined;
+}): InstallType | undefined {
+  const isInstalledPkg = !!installedPkg;
+  const currentPkgVersion = installedPkg?.attributes.version;
+  const lastStartedInstallVersion = installedPkg?.attributes.install_version;
+  if (installedPkg && pkgVersion === currentPkgVersion && pkgVersion !== lastStartedInstallVersion)
```

Comment:
If we can early do `if (!isInstalledPkg) return ...`, I think we can drop each of the `isInstalledPkg` checks afterwards